src/fsharp/service/ItemKey.fs

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.SourceCodeServices
+
+open System
+open System.IO
+open System.IO.MemoryMappedFiles
+open System.Reflection.Metadata
+open FSharp.NativeInterop
+open FSharp.Compiler
+open FSharp.Compiler.Range
+open FSharp.Compiler.Tast
+open FSharp.Compiler.Infos
+open FSharp.Compiler.NameResolution
+open FSharp.Compiler.AbstractIL.IL
+
+#nowarn "9"
+
+[<Sealed>]
+type ItemKeyStore(mmf: MemoryMappedFile, length) =
+
+    let mutable isDisposed = false
+    let checkDispose() =
+        if isDisposed then
+            raise (ObjectDisposedException("ItemKeyReader"))
```

Comment:
What is an `ItemKeyReader`? I can't find anything else named that in this PR

Reply:
Was the old name, it should be "ItemKeyStore".